OPERATING.md

```diff
@@ -0,0 +1,261 @@
+# Overview
+
+task-server is "fronted" by nginx
+
+API calls are made and tasks created/edited/removed within a SQL Database
+
+task-agent
+  Connects with a websocket to a task-server and is handed tasks to execute
+
+## task-server
+
+### configuration
+
+```
+log-level = 1
+addr = :8082
+db-type = mysql
+db-connect-str = DB_USERNAME:DB_PASSWORD@tcp(NGINX_REVERSE_PROXY:3306)/task_server?charset=utf8
+stats-enabled = true
+statsd-addr = localhost:8125
+statsd-type = standard
+admin-key = YOUR_TASK_SERVER_API_KEY
+exchange =
+#rabbitmq-url = amqp://RMQ_USER:RMQ_PASSWORD@NGINX_REVERSE_PROXY:5672/
+
+```
+## task-agent
+
+### configuration
+
+```
+log-level = 1
+server-url = wss://task-server.raintank.io/api/v1
+tsdb-url = https://tsdb-gw.raintank.io/
+snap-url = http://localhost:8181/
+stats-enabled = false
+statsd-addr = localhost:8125
+statsd-type = standard
+api-key = YOUR_TASK_SERVER_API_KEY
+name = task-agent-1
+```
+
+# Questions
+
+
+#### Registering an Agent
+For a task agent to work, it has to be pre-registered using the task-server API:
+
+http://localhost:4000/api/v1/agents
+
+CURL
+```
+curl -X POST \
+  http://localhost:4000/api/v1/agents \
+  -H 'Authorization: Bearer EASY' \
+  -H 'Cache-Control: no-cache' \
+  -H 'Content-Type: application/x-www-form-urlencoded' \
+  -H 'Postman-Token: 4ba0a028-5b71-f875-718b-63d71eb866ae' \
+  -H 'content-type: multipart/form-data; boundary=----WebKitFormBoundary7MA4YWxkTrZu0gW' \
+  -F name=agent2
+```
+
+HTTP/RAW
+```
+POST /api/v1/agents HTTP/1.1
+Host: localhost:4000
+Content-Type: multipart/form-data; boundary=----WebKitFormBoundary7MA4YWxkTrZu0gW
+Authorization: Bearer EASY
+Cache-Control: no-cache
+Postman-Token: aa131313-1086-75a1-7749-958479bd554f
+
+------WebKitFormBoundary7MA4YWxkTrZu0gW
+Content-Disposition: form-data; name="name"
+
+agent2
+------WebKitFormBoundary7MA4YWxkTrZu0gW--
+```
+
+GO Client:
+```
+package main
+
+import (
+	"fmt"
+	"strings"
+	"net/http"
+	"io/ioutil"
+)
+
+func main() {
```

Comment:
I dont think we need all of these Postman headers and complicated multipart form data.
lets just simplify this to
```
package main
import (
  "fmt"
  "net/http"
  "net/url"
  "io/ioutil"
)
func main() {
  payload := url.Values{}
  payload.Set('name', 'agent2')
  res, _ := http.DefaultClient.PostForm("http://localhost:4000/api/v1/agents", payload)
  defer res.Body.Close()
  body, _ := ioutil.ReadAll(res.Body)
  fmt.Println(res)
  fmt.Println(string(body))
}
```